### PR TITLE
MongoDB documentation improvements

### PR DESCRIPTION
**Pull request by loicmathieu**
Some improvements to the MongoDB documentation:

- Clarify multi-tenancy (see discussions https://github.com/quarkusio/quarkus/pull/7431#issuecomment-595826130)
- Add a note on the cryptic `Can't find a codec for class` error that should be fixed via `@RegisterForReflection`. Maybe this note needs to be on MongoDB with Panache as it was reported with the usage of MongoDB with Panache (all now normally fixed).
- Add some debugging tips

### CODE REVIEW

File: docs/src/main/asciidoc/mongodb.adoc [hunk 1]
```diff
@@ -236,6 +236,14 @@ Notice there's an extra bit in the key (the `users` and `inventory` segments).
 The syntax is as follows: `quarkus.mongodb.[optional name.][mongo connection property]`.
 If the name is omitted, it configures the default client.
 
+[NOTE]
+====
+Using multiple MongoDB clients allow to implements multi-tenancy for MongoDB by allowing to connect to multiple MongoDB clusters. +
```
Comment (geoand) [inline docs/src/main/asciidoc/mongodb.adoc:241]:
```suggestion
The use of multiple MongoDB clients enables multi-tenancy for MongoDB by allowing to connect to multiple MongoDB clusters. +
```

File: docs/src/main/asciidoc/mongodb.adoc [hunk 2]
```diff
@@ -236,6 +236,14 @@ Notice there's an extra bit in the key (the `users` and `inventory` segments).
...
+If you want to connect to multiple databases inside the same cluster,
+you can use the same client as a client is able to access all databases on the same cluster
```
Comment (geoand) [inline docs/src/main/asciidoc/mongodb.adoc:243]:
```suggestion
multiple clients are **not** necessary as a single client is able to access all databases in the same cluster
```

File: docs/src/main/asciidoc/mongodb.adoc [hunk 3]
```diff
@@ -598,6 +606,13 @@ Currently, Quarkus doesn't support the `mongodb+srv` protocol in native mode.
 link:https://docs.mongodb.com/manual/core/security-client-side-encryption/[Client-Side Field Level Encryption] is also not supported in native mode.
 ====
 
+[TIP]
+====
+If you encounter the following error when running your application in native mode: +
+`Failed to encode 'MyObject'. Encoding 'myVariable' errored with: Can't find a codec for class org.acme.MyVariable.` +
+This means that the `org.acme.MyVariable` class is not known to GraalVM, to fix this you can add the `@RegisterForReflection` annotation to your `MyVariable` class.
```
Comment (geoand) [inline docs/src/main/asciidoc/mongodb.adoc:613]: the remedy is to add the `@RegisterForReflection` annotation to your `MyVariable` class.

**Review by geoand — COMMENTED** [review]
Added some small suggestions

**Comment by loicmathieu** [comment]
@geoand I applied your suggestions

**Comment by geoand** [comment]
And I hadn't made any mistakes of my own :laughing: ?

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
Thanks for this, these are very good clarifications!

**Pull request merged**